erpnext/public/js/controllers/transaction.js

```diff
@@ -866,6 +867,10 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 				refresh_field('to_date');
 			}
 		}
+	},
+	
+	gross_profit: function(item) {
```

Comment:
set_gross_profit()

correct the formulae

and it is only applicable to sales order